lxd/storage/drivers/driver_pure_util.go

```diff
@@ -517,6 +572,475 @@ func (p *pureClient) disconnectHostFromVolume(poolName string, volName string, h
 	return nil
 }
 
+// getTarget retrieves the target that can be connected to using the initiator's qualified name.
+func (p *pureClient) getTarget() (*pureTarget, error) {
+	var resp pureResponse[pureTarget]
+
+	targetAddr := p.driver.config["pure.array.address"]
+
+	// Append service port to the address if missing.
+	// This way we can use exact matching in the filter to find the target.
+	if !strings.Contains(targetAddr, ":") {
+		// Default NVMe port is 4420.
```

Comment:
you've used switch cases elsewhere for this sort of logic, suggest doing the same here, with a default case that returns an error if mode is unhandled.